diff --git a/hybris/common/q/linker.cpp b/hybris/common/q/linker.cpp
index cf09de2..8ce6f18 100644
--- a/hybris/common/q/linker.cpp
+++ b/hybris/common/q/linker.cpp
@@ -119,6 +119,7 @@ static const char* const kAsanSystemLibDir    = "/data/asan/system/lib64";
 static const char* const kAsanOdmLibDir       = "/data/asan/odm/lib64";
 static const char* const kAsanVendorLibDir    = "/data/asan/vendor/lib64";
 static const char* const kRuntimeApexLibDir   = "/apex/com.android.runtime/lib64";
+static const char* const kI18nApexLibDir      = "/apex/com.android.i18n/lib64";
 #else
 static const char* const kSystemLibDir        = "/system/lib";
 static const char* const kOdmLibDir           = "/odm/lib";
@@ -127,6 +128,7 @@ static const char* const kAsanSystemLibDir    = "/data/asan/system/lib";
 static const char* const kAsanOdmLibDir       = "/data/asan/odm/lib";
 static const char* const kAsanVendorLibDir    = "/data/asan/vendor/lib";
 static const char* const kRuntimeApexLibDir   = "/apex/com.android.runtime/lib";
+static const char* const kI18nApexLibDir      = "/apex/com.android.i18n/lib"
 #endif
 
 static const char* const kAsanLibDirPrefix = "/data/asan";
@@ -135,6 +137,7 @@ static const char* const kDefaultLdPaths[] = {
   kSystemLibDir,
   kOdmLibDir,
   kVendorLibDir,
+  kI18nApexLibDir,
   nullptr
 };
 
@@ -145,6 +148,7 @@ static const char* const kAsanDefaultLdPaths[] = {
   kOdmLibDir,
   kAsanVendorLibDir,
   kVendorLibDir,
+  kI18nApexLibDir,
   nullptr
 };
 
